src/lang/builtins_bool.ml

```diff
@@ -27,9 +27,29 @@ let () =
   let register_op name op =
     add_builtin name ~cat:Bool ~descr:"Comparison of comparable values."
       [("", t, None, None); ("", t, None, None)] Lang.bool_t (fun p ->
-        let a = Lang.assoc "" 1 p |> Lang.demeth in
-        let b = Lang.assoc "" 2 p |> Lang.demeth in
-        Lang.bool (op (Lang.compare_values a b)))
+        let a = Lang.assoc "" 1 p in
+        let b = Lang.assoc "" 2 p in
+        let a' = Lang.demeth a in
+        let b' = Lang.demeth b in
+        (* For records, we also compare fields. *)
+        let ans =
+          if a'.Lang.value = Lang.Tuple [] && b'.Lang.value = Lang.Tuple [] then (
+            let r a =
+              let m, _ = Lang_values.V.split_meths a in
+              m
+            in
+            let a = r a in
+            let b = r b in
+            (* TODO: the order is not the expected one on records (for < we want
+               one field to be < and the other to be <=). *)
```

Comment:
What do you mean?

Reply:
Given records `r` and `s` with two fields `a` and `b`, the current implementations declares that `r < s` when both `r.a < s.a` and `r.b < s.b` whereas the natural order is rather
- `r.a < r.a` and `r.b <= s.b`, or
- `r.a <= r.a` and `r.b < s.b`

This could be fixed but I doubt that it's worth cluttering the code because I don't think that `<` comparison will be used that much on records...

Reply:
Works for me.